crates/dpapi-core/src/lib.rs

```diff
@@ -0,0 +1,22 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#![warn(clippy::std_instead_of_alloc)]
+#![warn(clippy::std_instead_of_core)]
+// #![warn(missing_docs)]
+
+extern crate alloc;
+
+mod decode;
+mod encode;
+mod error;
+pub mod gkdi;
+mod marker;
+mod padding;
+pub mod rpc;
```

Comment:
note: It seems `gkdi` and `rpc` modules are not providing core building blocks, and do not belong to the `dpapi-core` crate.

Reply:
I thought about it. I explored the IronRDP crates structure more and decided to move structures encoding and decoding implementations into a separate crate: `dpapi-pdu`. Moreover, I added an `alloc` feature to the `dpapi-core` crate and put the `EncodeVec` extension trait behind it.

I just pushed a new commit. Can you have a look?
Surprisingly, it was easy to refactor. I think it's a sign of a good architecture 🙃

Reply:
Thank you!!
It definitely is :wink: